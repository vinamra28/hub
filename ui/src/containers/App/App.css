.pf-c-page {
  position: fixed;
  width: 100%;
}
