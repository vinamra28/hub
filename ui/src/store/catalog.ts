import { types, Instance } from 'mobx-state-tree';

export const CatalogType = types
  .model({
    id: types.integer,
    name: types.string,
    selected: false
  })
  .actions((self) => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export type ICatalogType = Instance<typeof CatalogType>;
export type ICatalogStore = Instance<typeof CatalogStore>;

export const CatalogStore = types
  .model('CatalogStore', {
    list: types.array(CatalogType)
  })

  .views((self) => ({
    get count(): number {
      return self.list.length;
    },
    get catalogs(): Array<string> {
      const { list } = self;
      return list
        .filter((type: ICatalogType) => type.selected)
        .map((type: ICatalogType) => type.name);
    }
  }))

  .actions((self) => ({
    add(item: ICatalogType) {
      self.list.push(item);
    },
    clear() {
      self.list.map((item: ICatalogType) => (item.selected = false));
    }
  }));
