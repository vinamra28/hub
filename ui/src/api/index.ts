import axios from 'axios';
import { API_URL } from '../config/constants';
import { ICategory } from '../store/category';
import { IResource } from '../store/resource';

export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
}

export class Hub implements Api {
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resources() {
    try {
      return axios.get(`${API_URL}/resources`).then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }

  async resourceVersion(resourceId: string) {
    try {
      return axios
        .get(`${API_URL}/resource/${resourceId}/versions`)
        .then((response) => response.data);
    } catch (err) {
      return err.response;
    }
  }
}
