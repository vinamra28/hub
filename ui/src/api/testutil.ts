import { Api } from './';
import * as fs from 'fs';
import { ICategory } from '../store/category';
import { IResource } from '../store/resources';

export class FakeHub implements Api {
  dataDir: string;

  constructor(dataDir: string) {
    this.dataDir = dataDir;
  }

  async resources() {
    const data = `${this.dataDir}/resources.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<IResource>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }

  async categories() {
    const data = `${this.dataDir}/categories.json`;

    const ret = () => JSON.parse(fs.readFileSync(data).toString());
    return new Promise<ICategory>((resolve) => {
      setTimeout(() => resolve(ret()), 1000);
    });
  }
}
