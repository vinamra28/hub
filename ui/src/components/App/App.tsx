import React from 'react';
import { observer } from 'mobx-react';
import CategoryFilter from '../CategoryFilter/CategoryFilter';
import CatalogFilter from '../CatalogFilter/CatalogFilter';
import KindFilter from '../KindFilter/KindFilter';
import SortByFilter from '../SortByFilter/SortByFilter';
import { IResourceStore } from '../../store/resources';
import { Text, TextVariants, Grid, GridItem, Button } from '@patternfly/react-core';
import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';
import '@patternfly/react-core/dist/styles/base.css';

interface store {
  store: IResourceStore;
}

const App = observer(({ store }: store) => (
  <div className="App">
    <Grid
      hasGutter
      style={{
        marginTop: '5em'
      }}
    >
      <GridItem span={1} rowSpan={3}>
        <b
          style={{
            fontSize: '1.1em',
            verticalAlign: '-0.2em',
            color: '#484848',
            marginLeft: '3em',
            backgroundColor: 'white'
          }}
        >
          Sort
        </b>
      </GridItem>
      <GridItem span={1}>
        <SortByFilter store={store} />
      </GridItem>
      <Grid rows={2} rowSpan={2}>
        <GridItem rowSpan={2}>
          <Text
            component={TextVariants.h1}
            style={{
              fontWeight: 'bold',
              margin: '4.5em',
              fontSize: '1.1em',
              color: '#484848',
              marginBottom: '-2.0em',
              marginTop: '5em'
            }}
          >
            Refine By:
          </Text>
        </GridItem>
        <GridItem span={1} rowSpan={2}>
          <Button variant="plain" aria-label="Clear" onClick={store.clearAll}>
            <TimesIcon />
          </Button>
        </GridItem>
      </Grid>
      <CatalogFilter store={store.catalogStore} />
      <KindFilter store={store.kindStore} />
      <CategoryFilter store={store.categoryStore} />
    </Grid>
  </div>
));

export default App;
