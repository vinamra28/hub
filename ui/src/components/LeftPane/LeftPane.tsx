import React from 'react';
import { useObserver } from 'mobx-react';
import { GridItem, Grid, Text, TextVariants } from '@patternfly/react-core';
import SortByFilter from '../SortByFilter/SortByFilter';
import { IResourceStore } from '../../store/resources';
import CatalogFilter from '../CatalogFilter/CatalogFilter';
import KindFilter from '../KindFilter/KindFilter';
import CategoryFilter from '../CategoryFilter/CategoryFilter';

interface store {
  store: IResourceStore;
}

const LeftPane: React.FC<store> = (props: store) => {
  const store = props.store;
  return useObserver(() => (
    <div>
      <Grid sm={6} md={4} lg={3} xl2={1} style={{ marginLeft: '1.5em', marginTop: '2em' }}>
        <GridItem span={3}>
          <Text
            component={TextVariants.h1}
            style={{ fontWeight: 'bold', width: '3em', marginTop: '0.1em' }}
          >
            Sort
          </Text>
        </GridItem>
        <GridItem span={9}>
          <SortByFilter store={store} />
        </GridItem>
      </Grid>

      <KindFilter store={store.kindStore} />
      <CatalogFilter store={store.catalogStore} />
      <CategoryFilter store={store.categoryStore} />
    </div>
  ));
};

export default LeftPane;
