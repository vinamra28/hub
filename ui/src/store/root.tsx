import React, { ReactChild, ReactChildren } from 'react';
import { types, getEnv, Instance } from 'mobx-state-tree';
import { CategoryStore } from './category';
import { ResourceStore } from './resource';
import { Hub, Api } from '../api';

export const Root = types.model('Root', {}).views((self) => ({
  get api(): Api {
    return getEnv(self).api;
  },
  get categories() {
    return getEnv(self).categories;
  },
  get resources() {
    return getEnv(self).resources;
  }
}));

type IRoot = Instance<typeof Root>;

const initRootStore = (api: Api) => {
  const categories = CategoryStore.create({}, { api });
  const resources = ResourceStore.create({}, { api, categories });
  return Root.create({}, { api, categories, resources });
};

interface Props {
  children: ReactChild | ReactChildren;
}

let RootContext: React.Context<IRoot>;
export const useMst = () => React.useContext(RootContext);

export const createProviderAndStore = (api?: Api) => {
  const root = initRootStore(api || new Hub());
  RootContext = React.createContext(root);

  const Provider = ({ children }: Props) => (
    <RootContext.Provider value={root}> {children} </RootContext.Provider>
  );
  return { Provider, root };
};

export const createProvider = (api?: Api) => {
  const { Provider } = createProviderAndStore(api);
  return Provider;
};
