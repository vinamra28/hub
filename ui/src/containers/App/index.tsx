import React from 'react';
import { observer } from 'mobx-react';
import '@patternfly/react-core/dist/styles/base.css';
import { Grid, GridItem, Page, PageSection } from '@patternfly/react-core';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import LeftPane from '../../components/LeftPane';
import Background from '../../components/Background';
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import Resources from '../Resources';
import Details from '../Details';
import { createProvider } from '../../store/root';
import './App.css';

const Provider = createProvider();

const App: React.FC = observer(() => {
  return (
    <Provider>
      <Router>
        <Page header={<Header />} className="hub-page">
          <Route exact path="/" component={Background} />
          <PageSection>
            <Grid hasGutter>
              <GridItem span={12}>
                <Route exact path="/tekton/task/name/version" component={Details}></Route>
              </GridItem>
              <GridItem span={2}>
                <Route exact path="/" component={LeftPane}></Route>
              </GridItem>

              <GridItem span={10} rowSpan={1}>
                <Route exact path="/" component={Resources}></Route>
              </GridItem>
            </Grid>
          </PageSection>
          <Footer />
        </Page>
      </Router>
    </Provider>
  );
});

export default App;
