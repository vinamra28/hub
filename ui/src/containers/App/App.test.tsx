import React from 'react';
import App from '.';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore, createProvider } from '../../store/root';
import LeftPane from '../../components/LeftPane';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider } = createProviderAndStore(api);

describe('App', () => {
  it('should render the component correctly and match the snapshot', (done) => {
    const app = renderer.create(
      <Provider>
        <App />
      </Provider>
    );

    expect(app.toJSON()).toMatchSnapshot();
    done();
  });

  it('should find the component and match the count', () => {
    const Provider = createProvider();
    const component = mount(
      <Provider>
        <App />
      </Provider>
    );

    expect(component.find(LeftPane).length).toEqual(1);
  });
});
