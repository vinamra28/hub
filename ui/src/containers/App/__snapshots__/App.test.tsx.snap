// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component correctly and match the snapshot 1`] = `
Array [
  " ",
  " ",
  <div
    className="pf-c-page hub-page"
  >
    <header
      className="pf-c-page__header"
    >
      <div
        className="pf-c-page__header-brand"
      >
        <a
          className="pf-c-page__header-brand-link"
        >
          <img
            alt="Tekton Hub Logo"
            className="pf-c-brand"
            onClick={[Function]}
            src="logo.png"
          />
        </a>
      </div>
      <div
        className="pf-c-page__header-tools"
      >
        <div
          className="pf-l-grid"
        >
          <div
            className="pf-l-grid__item pf-m-11-col"
          >
            <input
              aria-invalid={false}
              aria-label="text input example"
              className="pf-c-form-control hub-search"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search for resources..."
              readOnly={false}
              required={false}
              spellCheck="false"
              type="search"
              value=""
            />
          </div>
        </div>
        <h3
          className=""
          data-pf-content={true}
        >
          Login
        </h3>
      </div>
    </header>
    <main
      className="pf-c-page__main"
      tabIndex={-1}
    >
      <div
        className="hub-background-image"
      >
        <div
          className="pf-c-content"
        >
          <h1
            className="hub-background-heading"
            data-pf-content={true}
          >
            Welcome to Tekton Hub
          </h1>
          <h3
            className="hub-tag-line"
            data-pf-content={true}
          >
            Discover, search and share reusable Tasks and Pipelines
          </h3>
        </div>
      </div>
      <section
        className="pf-c-page__main-section"
      >
        <div
          className="pf-l-grid pf-m-gutter"
        >
          <div
            className="pf-l-grid__item pf-m-12-col"
          />
          <div
            className="pf-l-grid__item pf-m-2-col"
          >
            <div
              className="pf-l-grid pf-m-gutter hub-leftpane"
            >
              <div
                className="pf-l-grid__item pf-m-3-col"
              >
                <h1
                  className="hub-leftpane-sort"
                  data-pf-content={true}
                >
                  Sort
                </h1>
              </div>
              <div
                className="pf-l-grid__item pf-m-9-col"
              >
                <div>
                  <div
                    className="pf-c-dropdown hub-sort-dropdown"
                    data-ouia-component-id={0}
                    data-ouia-component-type="PF4/Dropdown"
                    data-ouia-safe={true}
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="pf-c-dropdown__toggle"
                      disabled={false}
                      id="pf-dropdown-toggle-id-0"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <span
                        className="pf-c-dropdown__toggle-text"
                      >
                        Name
                      </span>
                      <span
                        className="pf-c-dropdown__toggle-icon"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 320 512"
                          width="1em"
                        >
                          <path
                            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            transform=""
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="pf-l-grid__item"
              >
                <div>
                  <div
                    className="pf-l-grid"
                  >
                    <div
                      className="pf-l-grid__item pf-m-6-col"
                    >
                      <h1
                        className="hub-filter-header"
                        data-pf-content={true}
                      >
                        Kind
                      </h1>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-3-col"
                    >
                      <button
                        aria-disabled={false}
                        aria-label="Clear"
                        className="pf-c-button pf-m-plain"
                        data-ouia-component-id={1}
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe={true}
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 352 512"
                          width="1em"
                        >
                          <path
                            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                            transform=""
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-12-col"
                    />
                  </div>
                </div>
              </div>
              <div
                className="pf-l-grid__item"
              >
                <div>
                  <div
                    className="pf-l-grid"
                  >
                    <div
                      className="pf-l-grid__item pf-m-6-col"
                    >
                      <h1
                        className="hub-filter-header"
                        data-pf-content={true}
                      >
                        Catalog
                      </h1>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-3-col"
                    >
                      <button
                        aria-disabled={false}
                        aria-label="Clear"
                        className="pf-c-button pf-m-plain"
                        data-ouia-component-id={2}
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe={true}
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 352 512"
                          width="1em"
                        >
                          <path
                            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                            transform=""
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-12-col"
                    />
                  </div>
                </div>
              </div>
              <div
                className="pf-l-grid__item"
              >
                <div>
                  <div
                    className="pf-l-grid"
                  >
                    <div
                      className="pf-l-grid__item pf-m-6-col"
                    >
                      <h1
                        className="hub-filter-header"
                        data-pf-content={true}
                      >
                        Category
                      </h1>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-3-col"
                    >
                      <button
                        aria-disabled={false}
                        aria-label="Clear"
                        className="pf-c-button pf-m-plain"
                        data-ouia-component-id={3}
                        data-ouia-component-type="PF4/Button"
                        data-ouia-safe={true}
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 352 512"
                          width="1em"
                        >
                          <path
                            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                            transform=""
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      className="pf-l-grid__item pf-m-12-col"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pf-l-grid__item pf-m-10-col pf-m-1-row"
          >
            <div
              className="pf-l-gallery pf-m-gutter hub-resource"
            >
              <span
                aria-valuetext="Loading..."
                className="pf-c-spinner pf-m-xl hub-spinner"
                role="progressbar"
              >
                <span
                  className="pf-c-spinner__clipper"
                />
                <span
                  className="pf-c-spinner__lead-ball"
                />
                <span
                  className="pf-c-spinner__tail-ball"
                />
              </span>
            </div>
          </div>
        </div>
      </section>
      <article
        className="pf-c-card hub-footer-card"
        data-ouia-component-id={4}
        data-ouia-component-type="PF4/Card"
        data-ouia-safe={true}
      >
        <div
          className="pf-l-grid"
        >
          <div
            className="pf-l-grid__item pf-m-12-col hub-footer-info"
          >
            <a
              href="https://cd.foundation"
              target="_"
            >
              <img
                alt="tekton.dev"
                src="https://tekton.dev/partner-logos/cdf.png"
              />
            </a>
          </div>
          <div
            className="pf-l-grid__item pf-m-12-col hub-footer-info"
          >
            <div
              className="pf-c-content hub-info-color"
            >
              <h3
                className=""
                data-pf-content={true}
              >
                Tekton is a
                 
                <a
                  className=""
                  data-pf-content={true}
                  href="https://cd.foundation"
                  target="_"
                >
                  Continuous Delivery Foundation
                </a>
                 
                project.
              </h3>
            </div>
          </div>
          <div
            className="pf-l-grid__item pf-m-12-col hub-logo-margin"
          >
            <img
              alt="Tekton"
              className="hub-logo-size"
              src="tekton.png"
            />
          </div>
          <div
            className="pf-l-grid__item pf-m-12-col hub-footer-description"
          >
            <p
              className=""
              data-pf-content={true}
            >
              © 2020 The Linux Foundation®. All rights reserved. The Linux Foundation has registered trademarks and uses trademarks. For a list of trademarks of The Linux Foundation, please see our
               
              <a
                className=""
                data-pf-content={true}
                href="https://www.linuxfoundation.org/trademark-usage/"
                target="_"
              >
                Trademark Usage page
              </a>
              . Linux is a registered trademark of Linus Torvalds.
               
              <a
                className=""
                data-pf-content={true}
                href="https://www.linuxfoundation.org/privacy/"
                target="_"
              >
                Privacy Policy
              </a>
               
              and
               
              <a
                className=""
                data-pf-content={true}
                href="https://www.linuxfoundation.org/terms/"
                target="_"
              >
                Terms of Use
              </a>
               
              .
            </p>
          </div>
        </div>
      </article>
    </main>
  </div>,
  " ",
  " ",
]
`;
